core/src/main/java/com/linecorp/armeria/common/ContentDisposition.java

```diff
@@ -0,0 +1,423 @@
+        this.type = type;
+        this.name = name;
+        this.filename = filename;
+        this.charset = charset;
+    }
+
+    /**
+     * Returns the disposition type, like for example {@code inline}, {@code attachment},
+     * {@code form-data}.
+     */
+    public String type() {
+        return type;
+    }
+
+    /**
+     * Returns the value of the {@code name} parameter, or {@code null} if not defined.
+     */
+    @Nullable
+    public String name() {
+        return name;
+    }
+
+    /**
+     * Returns the value of the {@code filename} parameter (or the value of the
+     * {@code filename*} one decoded as defined in the RFC 5987), or {@code null} if not defined.
+     */
+    @Nullable
+    public String filename() {
+        return filename;
+    }
+
+    /**
+     * Returns the charset defined in {@code filename*} parameter, or {@code null} if not defined.
+     */
+    @Nullable
+    public Charset charset() {
+        return charset;
+    }
+
+    private static List<String> tokenize(String headerValue) {
+        int index = headerValue.indexOf(';');
+        final String type = (index >= 0 ? headerValue.substring(0, index) : headerValue).trim();
+        checkArgument(!type.isEmpty(), "Content-Disposition header must not be empty");
+
+        final List<String> parts = new ArrayList<>(4);
+        parts.add(type);
+        if (index >= 0) {
+            do {
+                int nextIndex = index + 1;
+                boolean quoted = false;
+                boolean escaped = false;
+                while (nextIndex < headerValue.length()) {
+                    final char ch = headerValue.charAt(nextIndex);
+                    if (ch == ';') {
+                        if (!quoted) {
+                            break;
+                        }
+                    } else if (!escaped && ch == '"') {
+                        quoted = !quoted;
+                    }
+                    escaped = !escaped && ch == '\\';
+                    nextIndex++;
+                }
+                final String part = headerValue.substring(index + 1, nextIndex).trim();
+                if (!part.isEmpty()) {
+                    parts.add(part);
+                }
+                index = nextIndex;
+            }
+            while (index < headerValue.length());
+        }
+        return parts;
+    }
+
+    /**
+     * Decodes the given header field param as described in RFC 5987.
+     *
+     * <p>Only the US-ASCII, UTF-8 and ISO-8859-1 charsets are supported.
+     *
+     * @param filename the filename
+     * @param charset the charset for the filename
+     * @return the encoded header field param
+     * @throws UnsupportedEncodingException if the specified charset is not supported
+     *
+     * @see <a href="https://tools.ietf.org/html/rfc5987">RFC 5987</a>
+     */
+    private static String decodeFilename(String filename, Charset charset) throws UnsupportedEncodingException {
```

Comment:
Can remove `throws` clause because we're sure UTF-8 and ISO-8859-1 are always available.